Speed up casting by not killing apps known to be nonblocking

theychx opened the pull request:
Greetings! Just a little improvement to make catt faster

skorokithakis commented on catt/controllers.py:
```diff
@@ -66,8 +66,12 @@ def __init__(self):
+                             'BubbleUPnP']
+        if self.cast.app_display_name not in NON_BLOCKING_APPS:
+            self.kill()
+            while self.cast.app_display_name != 'Backdrop':
```
You're sure this updates every call, right?

skorokithakis commented:
Hello! Can you explain the rationale behind this a bit, and also how it works?

theychx commented:
> You're sure this updates every call, right?
Yep, tested it with serveral apps, it always gets the id right.

The point is that only a genuine "Custom Receiver" app (such as YouTube) might  exhibit the blocking behaviour. Theres no need to kill the currently running app and wait 5 seconds, if this app will let go of control voluntarily.
In the case of a  "Custom Receiver" app running, I changed the code so instead of just waiting 5 seconds, we poll for "Backdrop" to appear, and thus save some time here as well.

PS: I added BubbleUpnp to the list of nonblocking apps because it is an iteration of the "Styled Media Receiver" which I believe behaves just as the "Default Media Receiver"

skorokithakis commented:
Fantastic, thank you very much for the contribution!
